ttnn/ttnn/operations/conv2d.py

```diff
@@ -531,16 +531,15 @@ def conv2d(
         if config_shard_grid is not None:
             if config_shard_grid != input_shard_grid:
                 needs_reshard = True
-        input_height_sharded = input_shard_scheme == ttnn.TensorMemoryLayout.HEIGHT_SHARDED
-        if conv_config.height_sharding is not None:
-            if input_height_sharded != conv_config.height_sharding:
+        if conv_config.shard_layout is not None:
```

Comment:
when will conv2d be moved from py to c++?

Reply:
The latest version uses only C++. This [code path](https://github.com/tenstorrent/tt-metal/blob/8aa7d215f08c66d1b2cf413dcbe48e2675f0a9a5/ttnn/ttnn/operations/conv2d.py#L477) is a debug mode that compares the new C++ version with the old Python version, that was implemented during development of the new API.